### DISCUSSION TITLE
How can i create a contract/deploy using wagmi when i've abi and bytecode available?

### DISCUSSION THREAD

**Discussion by xsaad67** [wevm/wagmi#2216] [Q&A]
I've one smart contract abi through which i want the users to create contract. Is it possible using wagmi from ethers js i can only use 

```
import { ContractFactory } from 'ethers';
const factory = new ContractFactory(contractAbi, contractByteCode);

// If your contract requires constructor args, you can specify them here
const contract = await factory.deploy(deployArgs);

console.log(contract.address);
console.log(contract.deployTransaction);
```

how can i do the same through wagmi? Like i want to give the user functionality deploy a contract. Is it possible using Wagmi?

**Comment by tmm** [comment]
wagmi doesn't support this quite yet. once we move from ethers to [viem](https://viem.sh), we will add support.
